### PR TITLE
`Development`: Increase client test coverage for team owner search

### PR DESCRIPTION
**Pull request by akesfeden**
#### General

- [x] This is a small issue that I tested locally and was confirmed by pipeline tests.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

We would like to have a reasonably high test coverage percentage aimed to catch potential bugs before deployment.
### Description

Increase client test coverage of team owner search component over 90%.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage

|  File | Branches % | Lines % |
| --- | --- | --- |
| team-owner-search.component.ts: | 18.75% → 68.75% | 18.18% → 94.54% |

### CODE REVIEW

File: src/test/javascript/spec/component/team/team-owner-search.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    it('should initialize with team owner', () => {
+        comp.team = { owner };
+
+        fixture.detectChanges();
+
+        expect(comp.owner).toEqual(owner);
+        expect(comp.owner).not.toBe(owner); // Should be deep cloned
+
+        expect(comp.inputDisplayValue).toEqual(`${owner.name} (${owner.login})`);
```
Comment (b-fein) [inline src/test/javascript/spec/component/team/team-owner-search.component.spec.ts:41]:
```suggestion
        expect(comp.inputDisplayValue).toBe(`${owner.name} (${owner.login})`);
```

File: src/test/javascript/spec/component/team/team-owner-search.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        const searchText = owner.login!;
+
+        comp.course = { id: 1 };
+
+        let onSearchResult: User[] | undefined = undefined;
+        comp.onSearch(of(searchText)).subscribe((result) => (onSearchResult = result));
+
+        expect(searchFailedSpy).toBeCalledTimes(1);
+        expect(searchFailedSpy).toBeCalledWith(false);
```
Comment (b-fein) [inline src/test/javascript/spec/component/team/team-owner-search.component.spec.ts:60]:
```suggestion
        expect(searchFailedSpy).toHaveBeenCalledTimes(1);
        expect(searchFailedSpy).toHaveBeenCalledWith(false);
```
The assertion is the same with just another name but we usually only use this variant.

**Review by b-fein — DISMISSED** [review]
Looks good, only two small suggestions about using different assertions.

File: src/test/javascript/spec/component/team/team-owner-search.component.spec.ts (hunk 2 above)
Comment (akesfeden) [reply src/test/javascript/spec/component/team/team-owner-search.component.spec.ts]: Thank you @b-fein, I applied your suggestions.

Comment (b-fein) [reply src/test/javascript/spec/component/team/team-owner-search.component.spec.ts]: Thanks. There are a few more of those different names in this file and I didn’t want to mark all of them, though. Sorry I wasn’t making that clear in my comment.

Comment (akesfeden) [reply src/test/javascript/spec/component/team/team-owner-search.component.spec.ts]: Oh sorry. I now changed all occurrences of `.toBeCalled*` methods to `.toHaveBeenCalled*`.

**Review by b-fein — APPROVED** [review]
Thanks! Code looks good. :+1:

File: src/test/javascript/spec/component/team/team-owner-search.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [TeamOwnerSearchComponent, MockDirective(NgbTypeahead), MockDirective(NgModel), MockPipe(ArtemisTranslatePipe)],
+            providers: [{ provide: CourseManagementService, useClass: MockCourseManagementService }],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(TeamOwnerSearchComponent);
+        comp = fixture.componentInstance;
+        courseService = fixture.debugElement.injector.get(CourseManagementService);
```
Comment (bassner) [inline src/test/javascript/spec/component/team/team-owner-search.component.spec.ts:30]: Should we use ``TestBed.inject()`` here for brevity?

Comment (akesfeden) [reply src/test/javascript/spec/component/team/team-owner-search.component.spec.ts:30]: With the current component under test that would also be fine I guess, but according to our previous discussion at https://github.com/ls1intum/Artemis/pull/4686#discussion_r808212210 I am still a bit reluctant that this brevity is worth having the low but extra risk. Maybe we can decide on a convention for injections in general.

Comment (bassner) [reply src/test/javascript/spec/component/team/team-owner-search.component.spec.ts:30]: Well, I think, as long as it works, what's wrong about keeping it short? You would notice if it doesn't. In fact, we do use this longer, more verbose method 283 times in our test code, but ``TestBed.inject`` is used 684 times without breaking anything.
But you're right, we don't have guidelines on this, and this works, so we can keep it

**Review by bassner — APPROVED** [review]

**Pull request merged**